pkg/util/helper/binding_test.go

```diff
@@ -180,3 +182,96 @@ func TestObtainBindingSpecExistingClusters(t *testing.T) {
 		})
 	}
 }
+
+func TestIsBindingScheduled(t *testing.T) {
+	tests := []struct {
+		name   string
+		status *workv1alpha2.ResourceBindingStatus
+		want   bool
+	}{
+		{
+			name: "condition type is not Scheduled",
+			status: &workv1alpha2.ResourceBindingStatus{
+				Conditions: []metav1.Condition{
+					{Type: clusterapis.ClusterConditionReady, Status: metav1.ConditionTrue},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "condition status is not true",
+			status: &workv1alpha2.ResourceBindingStatus{
+				Conditions: []metav1.Condition{
+					{Type: workv1alpha2.Scheduled, Status: metav1.ConditionTrue},
+				},
+			},
+			want: true,
+		},
+		{
+			name: "condition status is not true",
+			status: &workv1alpha2.ResourceBindingStatus{
+				Conditions: []metav1.Condition{
+					{Type: workv1alpha2.Scheduled, Status: metav1.ConditionFalse},
+				},
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsBindingScheduled(tt.status); got != tt.want {
+				t.Errorf("IsBindingScheduled() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSortClusterByWeight(t *testing.T) {
+	tests := []struct {
+		name string
+		m    map[string]int64
+		want ClusterWeightInfoList
+	}{
+		{
+			name: "different weight",
+			m: map[string]int64{
+				"b": 1,
+				"a": 2,
+			},
+			want: []ClusterWeightInfo{
+				{
+					ClusterName: "a",
+					Weight:      2,
+				},
+				{
+					ClusterName: "b",
+					Weight:      1,
+				},
+			},
+		},
+		{
+			name: "same weight",
+			m: map[string]int64{
+				"b": 1,
+				"a": 1,
+			},
+			want: []ClusterWeightInfo{
+				{
+					ClusterName: "a",
+					Weight:      1,
+				},
+				{
+					ClusterName: "b",
+					Weight:      1,
+				},
+			},
```

Comment:
The `map` is unordered, which means we can't assume always get `a` then get `b`.  
I don't know how to test the `same weight` scenario.